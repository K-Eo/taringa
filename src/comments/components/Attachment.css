.Attachment-iframe-wrapper {
  height: 0;
  padding-bottom: 56%;
  position: relative;
  width: 100%;
}

.Attachment-iframe {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
}
