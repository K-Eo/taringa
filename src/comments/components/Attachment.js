import React from 'react'

import './Attachment.css'

const ImageAttachment = ({ url }) => {
  return <img className="img-fluid" src={url} alt={url} />
}

const LinkAttachment = ({ videoId }) => {
  const src = `https://www.youtube.com/embed/${videoId}`

  return (
    <div className="Attachment-iframe-wrapper">
      <iframe
        allow="autoplay; encrypted-media"
        className="Attachment-iframe"
        allowFullScreen
        frameBorder="0"
        height="720"
        src={src}
        title={src}
        width="1280"
      />
    </div>
  )
}

const Attachment = ({ type, ...rest }) => {
  switch (type) {
    case 'image':
      return <ImageAttachment {...rest} />
    case 'link':
      return <LinkAttachment {...rest} />
    default:
      return <div>Attachment no supported</div>
  }
}

export default Attachment
