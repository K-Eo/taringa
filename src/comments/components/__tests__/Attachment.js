import React from 'react'

import Attachment from '../Attachment'
import { ImageAttachment, LinkAttachment } from '../Attachment'

describe('Attachment', () => {
  it('exists', () => {
    expect(Attachment).toBeDefined()
  })

  it('renders default', () => {
    const wrapper = shallow(<Attachment />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders image attachment', () => {
    const props = {
      type: 'image',
      url: 'https://placehold.it/300',
    }

    const wrapper = shallow(<Attachment {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders link attachment', () => {
    const props = {
      type: 'link',
      videoId: 'foobar',
    }

    const wrapper = shallow(<Attachment {...props} />)

    expect(wrapper).toMatchSnapshot()
  })
})

describe('Image Attachment', () => {
  it('exists', () => {
    expect(ImageAttachment).toBeDefined()
  })

  it('renders default', () => {
    const wrapper = shallow(<ImageAttachment />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders image', () => {
    const wrapper = shallow(<ImageAttachment url="https://placehold.it/300" />)

    expect(wrapper).toMatchSnapshot()
  })
})

describe('Link Attachment', () => {
  it('exists', () => {
    expect(LinkAttachment).toBeDefined()
  })

  it('renders default', () => {
    const wrapper = shallow(<LinkAttachment />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders video', () => {
    const wrapper = shallow(<LinkAttachment videoId="foobar" />)

    expect(wrapper).toMatchSnapshot()
  })
})
