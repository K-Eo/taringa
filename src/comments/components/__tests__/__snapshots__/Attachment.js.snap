// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attachment renders default 1`] = `
<div>
  Attachment no supported
</div>
`;

exports[`Attachment renders image attachment 1`] = `
<ImageAttachment
  url="https://placehold.it/300"
/>
`;

exports[`Attachment renders link attachment 1`] = `
<LinkAttachment
  videoId="foobar"
/>
`;

exports[`Image Attachment renders default 1`] = `""`;

exports[`Image Attachment renders image 1`] = `
<img
  alt="https://placehold.it/300"
  className="img-fluid"
  src="https://placehold.it/300"
/>
`;

exports[`Link Attachment renders default 1`] = `""`;

exports[`Link Attachment renders video 1`] = `
<div
  className="Attachment-iframe-wrapper"
>
  <iframe
    allow="autoplay; encrypted-media"
    allowFullScreen={true}
    className="Attachment-iframe"
    frameBorder="0"
    height="720"
    src="https://www.youtube.com/embed/foobar"
    title="https://www.youtube.com/embed/foobar"
    width="1280"
  />
</div>
`;
